### DISCUSSION TITLE
How to add relations buried in an embedded entity?

### DISCUSSION THREAD

**Discussion by pedramardakani** [ppetzold/nestjs-paginate#831] [Q&A] [answered]
Hi there. I have been trying to make this work, but no luck so far.

I want to create an `invoice`, which might assign an `inspector` who would approve the `quantity` of things inside this invoice. The structure below should reflect that:

> The `Invoice` entity, contains an embedded `quantity` entity of type `Inspection`, which holds a `Many-to-One` relation with the `User` entity. 

A minimal preview would be like the following:

```typescript
@Entity()
export class User {
  @PrimaryGeneratedColumn('uuid')
  uuid: string

  @Column()
  name: string

  // other fields
}

export abstract class Inspection {
  @ManyToOne(() => User)
  inspector?: User;

  // other fields 
}

@Entity()
export class Invoice {
  @PrimaryGeneratedColumn('uuid')
  uuid: string

  @Column(() => Inspection)
  quantity?: Inspection

  // other fields
}
```

Now, let's say I want to get the invoice, and load the `inspector` information inside (e.g. `name`) as well. It is possible to use `TypeORM` relations like so:

```typescript
const id = "some-valid-uuid-here"
async findOne(id: string) {
  return this.invoiceRepository.findOne({ where: { uuid: id }, relations: ['quantity.inspector'] })
}
```

But I can't get it to work with same relation specification inside paginate config. For instance:

```typescript
static PAGINATION_CONFIG: PaginateConfig<Invoice> = {
  // other settings with filter, sortable columns, etc.

  relations: ['quantity.inspector'],
  // P.S the form below doesn't work as well
  // relations: { quantity: { inspector: true } }
```

It raises the error:

> [Nest] 912690  - 12/12/2023, 3:25:40 PM   ERROR [ExceptionsHandler] Relation with property path quantity in entity was not found.
TypeORMError: Relation with property path quantity in entity was not found.
    at JoinAttribute.getValue (/path/to/project/src/query-builder/JoinAttribute.ts:187:23)
    at JoinAttribute.get relation [as relation] (/path/to/project/src/query-builder/JoinAttribute.ts:191:53)
    at JoinAttribute.get metadata [as metadata] (/path/to/project/src/query-builder/JoinAttribute.ts:203:18)
    at SelectQueryBuilder.join (/path/to/project/src/query-builder/SelectQueryBuilder.ts:2038:53)
    at SelectQueryBuilder.leftJoin (/path/to/project/src/query-builder/SelectQueryBuilder.ts:467:14)
    at SelectQueryBuilder.leftJoinAndSelect (/path/to/project/src/query-builder/SelectQueryBuilder.ts:607:14)
    at /path/to/project/node_modules/nestjs-paginate/src/paginate.ts:226:30
    at Array.forEach (<anonymous>)
    at createQueryBuilderRelations (/path/to/project/node_modules/nestjs-paginate/src/paginate.ts:222:36)
    at paginate (/path/to/project/node_modules/nestjs-paginate/src/paginate.ts:236:9)

P.S:

I have read [usage-with-relations](https://github.com/ppetzold/nestjs-paginate#usage-with-relations), I know about the merged pull request [Embedded entities #369](https://github.com/ppetzold/nestjs-paginate/pull/369), I have the latest `nestjs-paginate` installed, i.e `8.6.0`.

**Comment by ppetzold** [comment, marked as answer]
umm.. embedded entities don't need any relation definition - see https://github.com/ppetzold/nestjs-paginate/blob/master/src/paginate.spec.ts#L1235

both notations `relations: ['quantity.inspector']` and `relations: { quantity: { inspector: true } }` perform actual table joins, that's why you getting the error above.

Your case where an embedded entity holds another relation is simply an edge case where this lib has no coverage yet - and hence the implementation didn't consider this.

You are welcome to submit a bug fix :)

**Reply by pedramardakani** [reply]
Woah thank you for the quick answer! Sure, I'll try doing that :+1:
